examples/basic_ibe/frontend/src/main.ts

```diff
@@ -226,33 +226,13 @@ async function displayMessages(inbox: Inbox) {
                     if ("Err" in result) {
                         alert("Error deleting message: " + result.Err);
                     } else {
-                        // Remove the message element from the DOM
-                        const messageElement = target.closest(".message");
-                        if (messageElement) {
-                            messageElement.remove();
-
-                            // If this was the last message, show the "no messages" message
-                            const messagesDiv =
-                                document.getElementById("messages")!;
-                            if (messagesDiv.children.length === 0) {
-                                const noMessagesDiv =
-                                    document.createElement("div");
-                                noMessagesDiv.className = "no-messages";
-                                noMessagesDiv.textContent =
-                                    "No messages in the inbox.";
-                                messagesDiv.appendChild(noMessagesDiv);
-                            }
-                        }
+                        // Re-load all messages to refresh message indices
+                        await showMessages();
```

Comment:
This fixes a bug that was already there before this PR. We need to reload all messages because the frontend knows messages only by index, but the indices actually change when a message is deleted.